state-chain/runtime/src/lib.rs

```diff
@@ -1745,7 +1745,8 @@ impl_runtime_apis! {
 				let execute_at = core::cmp::max(block, current_block.saturating_add(1));
 
 				let swaps: Vec<_> = swaps_for_block.iter().filter(|swap| swap.from == base_asset || swap.to == base_asset).cloned().collect();
-				Swapping::get_scheduled_swap_legs(swaps, base_asset).unwrap().into_iter().map(move |swap| (swap, execute_at))
+				// TODO: Use the FoK logic when this fails instead of unwrap_or_default.
```

Comment:
can we create a ticket for this and put it in the FoK project

Reply:
Yeah, we should return the swaps for the first leg.

Reply:
PRO-1437

Reply:
> [PRO-1437](https://linear.app/chainflip/issue/PRO-1437)

and in the meantime we will return the first leg, or nothing? Either way, I think that ticket should be done soon anyway

Reply:
Not sure - yes first leg is better than nothing. But it would be good to communicate somehow that the second legs are missing.

Reply:
I did think of that but it seemed quite complicated. @msgmaxim do you think it's doable?

Reply:
I like the suggestion in [PRO-1437](https://linear.app/chainflip/issue/PRO-1437). Let's use price estimates for calculating the second leg. (I'm not yet sure whether we shoud alwasy do it this way, or only fall back when the simulation fails; both are estimates anyway, so I'm leaning towards the simpler approach).

I don't mind merging this (after we fix fix not communicating the first leg). However, we might save a bit of time if we implement [PRO-1437](https://linear.app/chainflip/issue/PRO-1437) directly.

Reply:
I agree the simpler approach is better. LPs are unlikely to blindly use our output values directly anyway: they will (or at least, they *should*) decide on their own price and apply that.